uk.co.compucorp.civicrm.appraisals/js/src/appraisals/controllers/appraisals-dashboard-ctrl.js

```diff
@@ -74,49 +75,56 @@ define([
              * It can either add a new page worth of cycles to the existing list
              * or it can reset the entire list (in case new filters have been chosen)
              *
+             * After retrieving the list of cycles, it also requests the status
+             * overview for those cycles in the current week (ONLY for first page)
+             *
              * @param {boolean} addPage - If it's to request the next page
              */
             vm.requestCycles = function (addPage) {
-                if (addPage && vm.loadingDone) {
+                var cycles;
+
+                if (addPage && vm.loading.done) {
                     return;
                 }
 
-                pagination.page = !!addPage ? pagination.page + 1 : 1;
-
-                AppraisalCycle.all(filters(), pagination).then(function (cycles) {
-                    if (addPage) {
-                        cycles.list.forEach(function (cycle) {
-                            vm.cycles.push(cycle);
-                        });
-                    } else {
-                        vm.cycles = cycles.list;
-                    }
-
-                    vm.loadingDone = vm.cycles.length === cycles.total;
-                });
+                vm.loading.inProgress = true;
+                vm.pagination.page = !!addPage ? vm.pagination.page + 1 : 1;
```

Comment:
Well not strictly, but given that it's a parameter passed to the function i think it's better to always force it to be a boolean